### ISSUE TITLE
Broken Stacked Bar Chart

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#1805]
```
{
  "data": {"url": "data/movies.json"},
  "mark": "area",
  "encoding": {
    "x": {
      "bin": {"maxbins": 10},
      "field": "IMDB_Rating",
      "type": "quantitative"
    },
    "color": {
      "field": "Source",
      "type": "nominal"
    },
    "y": {
      "aggregate": "count",
      "field": "*",
      "type": "quantitative"
    }
  }
}
```
![vega_editor](https://cloud.githubusercontent.com/assets/111269/21829722/b1b7d208-d74c-11e6-83b6-346c40a22855.png)

**Comment by domoritz** [comment]
#Accidentalart

**Comment by kanitw** [comment]
Actually it looks a bit better now.



The culprit is that we need to impute and stack by both bin_start and bin_end and apply collect transform after stacking instead of before stacking.  

```json
  {
      "name": "stacked",
      "source": "summary",
      "transform": [
        {
          "type": "impute",
          "field": "count_*",
          "groupby": ["Source"],
          "orderby": ["bin_IMDB_Rating_start", "bin_IMDB_Rating_end"],  **
          "method": "value",
          "value": 0
        },
        {
          "type": "stack",
          "groupby": ["bin_IMDB_Rating_start", "bin_IMDB_Rating_end"],  **
          "field": "count_*",
          "sort": {"field": ["Source"],"order": ["descending"]},
          "as": ["count_*_start","count_*_end"],
          "offset": "zero"
        },
        {
          "type": "collect",
          "sort": {
            "field": "bin_IMDB_Rating_start",
            "order": "descending"
          }
        }
      ]
    }
```

![vega_editor](https://cloud.githubusercontent.com/assets/111269/21859143/63b5f3ac-d7e0-11e6-9006-b1ae05f7cebd.png)

**Comment by kanitw** [comment]
Fixed in #1819
